symengine/dict.h

```diff
@@ -185,6 +185,165 @@ std::ostream &operator<<(std::ostream &out, const SymEngine::vec_basic &d);
 std::ostream &operator<<(std::ostream &out, const SymEngine::set_basic &d);
 std::ostream &operator<<(std::ostream &out, const SymEngine::map_int_Expr &d);
 
+// dict wrapper
+template <typename Key, typename Value, typename Wrapper>
+class ODictWrapper
```

Comment:
It can act like anything. The actual wrappers which behave as the polynomial are `UIntDict` and `UnivariateExprPolynomial` which are derived, and present in the polynomial.h file. But I'll move it to the polynomial file if you think that's right.